src/submission/models.py

```diff
@@ -76,6 +81,67 @@ def article_media_upload(instance, filename):
     return os.path.join(path, filename)
 
 
+CREDIT_ROLES = {
+    "Conceptualization": (
+        "Conceptualization",
+        "https://credit.niso.org/contributor-roles/conceptualization/",
+    ),
+    "Data Curation": (
+        "Data Curation",
+        "https://credit.niso.org/contributor-roles/data-curation/",
+    ),
+    "Formal Analysis": (
+        "Formal Analysis",
+        "https://credit.niso.org/contributor-roles/formal-analysis/",
+    ),
+    "Funding Acquisition": (
+        "Funding Acquisition",
+        "https://credit.niso.org/contributor-roles/funding-acquisition/",
+    ),
+    "Investigation": (
+        "Investigation",
+        "https://credit.niso.org/contributor-roles/investigation/",
+    ),
+    "Methodology": (
+        "Methodology",
+        "https://credit.niso.org/contributor-roles/methodology/",
+    ),
+    "Project Administration": (
+        "Project Administration",
+        "https://credit.niso.org/contributor-roles/project-administration/",
+    ),
+    "Resources": (
+        "Resources",
+        "https://credit.niso.org/contributor-roles/resources/",
+    ),
+    "Software": (
+        "Software",
+        "https://credit.niso.org/contributor-roles/software/",
+    ),
+    "Supervision": (
+        "Supervision",
+        "https://credit.niso.org/contributor-roles/supervision/",
+    ),
+    "Validation": (
+        "Validation",
+        "https://credit.niso.org/contributor-roles/validation/",
+    ),
+    "Visualization": (
+        "Visualization",
+        "https://credit.niso.org/contributor-roles/visualization/",
+    ),
+    "Writing - Original Draft": (
+        "Writing - Original Draft",
+        "https://credit.niso.org/contributor-roles/writing-original-draft/",
+    ),
+    "Writing - Review & Editing": (
+        "Writing - Review & Editing",
+        "https://credit.niso.org/contributor-roles/writing-review-editing/",
+    ),
+}
+
+CREDIT_ROLE_CHOICES = [(key, values[0]) for key, values in CREDIT_ROLES.items()]
```

Comment:
As mentioned in my last review, this still seems unnecessarily complicated to me. Couldn't we have a simpler choices set like this?

```py
CREDIT_ROLE_CHOICES = [
    ("conceptualization", "Conceptualization"),
    ("data-curation", "Data Curation"),
    ("formal-analysis", "Formal Analysis"),
    ("funding-acquisition", "Funding Acquisition"),
    ...
    ("writing-review-editing": "Writing - Review & Editing")
]
```

Then to get the URI form for downstream metadata, a method:

```py
class CreditRecord(AbstractLastModifiedModel):
    def uri(self):
        return f"https://credit.niso.org/contributor-roles/{self.role}/"
```

And for the string representation, we can get the choice label:

```py
    def __str__(self):
        return self.get_role_display()
```

Reply:
The reason to not tie together the URL to a given pattern based on key, was that we anticipate additional roles to be added that do not have an entry in credit.niso.org

In fact, in Martin's use case, there were already a few entries that were custom IIRC

As for using the niso slugs for the credit roles, that's fine. I think the only prior objection against using the actual URL as the database keys/identifiers.